crates/cxx-qt-gen/src/generator/cpp/signal.rs

```diff
@@ -82,12 +91,42 @@ pub fn generate_cpp_signals(
                 }}
                 "#,
                 ident = signal_ident,
-                parameters = parameter_types_emitter.join(", "),
-                parameter_values = parameter_values.join(", "),
+                parameters = parameter_types_rust.join(", "),
+                parameter_values = parameter_values_emitter.join(", "),
                 emit_ident = emit_ident,
                 qobject_ident = qobject_ident,
             },
         });
+
+        // Generate connection
+        parameter_types_rust.insert(0, format!("{qobject_ident}&"));
+        parameter_values_connection.insert(0, "*this".to_owned());
+
+        generated.methods.push(CppFragment::Pair {
+            header: format!(
+                "::std::unique_ptr<QMetaObject::Connection> {connect_ident}(::rust::Fn<void({parameters})> func, Qt::ConnectionType type);",
+                parameters = parameter_types_rust.join(", ")
+            ),
+            source: formatdoc! {
+                r#"
+                ::std::unique_ptr<QMetaObject::Connection>
+                {qobject_ident}::{connect_ident}(::rust::Fn<void({parameters_rust})> func, Qt::ConnectionType type)
+                {{
+                    return ::std::make_unique<QMetaObject::Connection>(QObject::connect(this,
+                            &{qobject_ident}::{signal_ident},
+                            this,
```

Comment:
This syntax doesn't provide a way to specify anything other than `this` for the target. Is that a problem?

Reply:
Right this is just for adding a function pointer to a signal, if you want to call a slot/method on another object then you should
- store a pointer to the object as a property/field
- connect to the signal you are interested in
- in the function pointer retrieve the qobject
- call the slot/invokable on the qobject via emit(...) or the method name

I think trying to add syntax and safety around making a direct connection between two objects is way more tricky to do and should be a separate task (if we even wanted it), and this is a good enough workaround for now.

Reply:
Can you make an example of this workaround to ensure that it does work before we commit to this syntax?

Reply:
Added to the nested objects an example causing `OuterObject::Called` to trigger `InnerObject::Called`